crates/cli/src/tasks/rust.rs

```diff
@@ -28,22 +29,36 @@ pub(crate) fn build_rust(project_path: &Path, skip_clippy: bool, build_debug: bo
         );
     }
 
-    // Note: Clippy has to run first so that it can build & cache deps for actual build while checking in parallel.
     if !skip_clippy {
-        let clippy_conf_dir = tempfile::tempdir()?;
-        fs::write(clippy_conf_dir.path().join("clippy.toml"), CLIPPY_TOML)?;
-        eprintln!("checking crate with spacetimedb's clippy configuration");
-        let out = cargo_cmd(
-            "clippy",
-            build_debug,
-            &["--", "--no-deps", "-Aclippy::all", "-Dclippy::disallowed-macros"],
-        )
-        .dir(project_path)
-        .env("CLIPPY_DISABLE_DOCS_LINKS", "1")
-        .env("CLIPPY_CONF_DIR", clippy_conf_dir.path())
-        .unchecked()
-        .run()?;
-        anyhow::ensure!(out.status.success(), "clippy found a lint error");
+        let mut err_count: u32 = 0;
+        for file in walkdir::WalkDir::new(project_path).into_iter() {
```

Comment:
should we be walking the whole project directory? As-is, this erroneously catches some `println`s in BitCraft's `build.rs`. Should we be hardcoding `src/` or something...?

Reply:
There does not appear to be a way to use e.g. `cargo` to list all the source files, as far as I can find. It looks like the best we could do is parse the output of the cargo build in order to figure out which files were involved.

Reply:
Alternatively, we could add an extra argument that defaults to `src/`, e.g. `--lint-path` or something.